chess-AI/train.py

```diff
@@ -155,22 +158,46 @@ def training_episode(self, nnet, games, epochs, batch_size):
 
                         # Reset gradients
                         opt.zero_grad()
-
                 print(losses)
-
+            # Saves model to specified file, or a new file if not specified.
+            if self.save_path != None:
+                save_model(nnet, self.save_path)
+            else:
+                for i in range(num_saved_models):
+                    if not(os.path.isfile(f'chess-AI/models-{i+1}.pt')):
+                        if overwrite_save and i != 0:
+                            save_model(nnet, f'chess-AI/models-{i}.pt')
+                            break
+                        save_model(nnet, f'chess-AI/models-{i+1}.pt')
+                        break
+                    if i == num_saved_models - 1:
+                        save_model(nnet, f'chess-AI/models-{num_saved_models}.pt')
```

Comment:
Would it be possible to have all this logic in the save function itself? I imagine this is something that might be used for stockfishtrain as well.